python/dgl/data/adapter.py

```diff
@@ -114,6 +123,8 @@ def process(self):
                 print('Generating train/val/test masks...')
             utils.add_nodepred_split(self, self.split_ratio, self.target_ntype)
```

Comment:
Why not putting the logic to the `add_nodepred_split` function?

Reply:
in order to share among different code path of `split_ratio`. and shared among `process` and `load`. what's more, as this api(`set_split_index`) amis to add attributes into class, I think it's better and clearer to be a method of dataset, instead of a utility function.